bee-protocol/src/workers/peer/manager.rs

```diff
@@ -187,6 +189,16 @@ where
                             info!("Disconnected peer {}.", peer.0.alias());
                         })
                         .unwrap_or_default(),
+                    NetworkEvent::PeerUnreachable { peer_id, peer_info } => {
```

Comment:
Why are we delegating this to the protocol ? We're just sending a command back to the gossip so why can't it be all done internally ?

Reply:
Because how manual peers and auto peers are treated may be a higher level protocol decision. It's just reporting back whether establishing a gossip connection with a peer (as ordered by the bee-protocol) succeeded or not. In this case it was ordered to establish a connection to `peer_id`, it tried a few times, and then reports back that the peer is ... well, unreachable at the moment. `bee-protocol` can now decide what to do with that information. For discovered peers it makes sense to tell `bee-autopeering` to remove that peer from the neighborhood, for "manual" it may decide to log a warning about a configuration issue. Since `bee-gossip` and `bee-autopeering` don't know about each other for good reason (it would be wrong to make `bee-gossip` dependent on `bee-autopeering`), we need to delegate this to a higher level (`bee-protocol`).

At least that's how I see how these things should be architectured. Doesn't mean I am right of course.